services/motion/builtin/builtin_test.go

```diff
@@ -408,7 +408,9 @@ func TestObstacleReplanningSlam(t *testing.T) {
 		ExecutionID:   executionID,
 		LastPlanOnly:  true,
 	})
-	test.That(t, err, test.ShouldBeNil)
+	if err != nil {
+		test.That(t, strings.Contains(err.Error(), motion.ErrGoalWithinPlanDeviation.Error()), test.ShouldBeTrue)
```

Comment:
I dont like that we have to catch this case manually. Either this is an error and it should be treated as one, or its not an error and we shouldnt be returning it as an error

Reply:
There are two cases to consider here.

Case1 is where the robot starts within planDevMM of the destination. In this case, I think it makes sense to return an error.

Case2 is what we are dealing with here. It is where the robot does not start within planDevMM of the destination and is also polling for transient obstacles. Suppose that when we are within planDevMM of the destination we continue polling so we enter `obstaclesIntersectPlan` and find that we have a collision. This will end up returning an error which will then trigger a replan event which means we will make our way through `createBaseMoveRequest` again. Since, at this point in time, we are already within planDevMM we will end up returning an error, when I believe we should not.
To avoid returning an error in this case we should check if we are within planDevMM of the destination if `obstaclesIntersectPlan` returns an error. If we are, then we should not return an error and instead issue a warning. If we are not within planDevMM of the destination then we will return error and trigger a replan.

Reply:
I don't think we should be treating this as an error at all then. It's the success criteria

Reply:
I'm concerned that this test may not be operating as expected, if case 2 is what we are seeing. If we are so very close to the goal, then the path should be nearly done, and we would not expect the path to intersect any obstacles.

I actually wonder if this error case should be fully removed. I think it's reasonable to ask a base to attempt to fine-tune its position with poses smaller than plan deviation.

Reply:
https://viam.atlassian.net/browse/RSDK-8492

Reply:
I am wondering if we should also skip this test so that we do not encounter a failure in GitHub CI.

Reply:
Could you elaborate on why the solution would be involved?

IMO we should just replan on an intersection error always, and we should create a plan always regardless of how close the base starts to the goal; if it only needs to drive 2mm so be it.

Reply:
> Could you elaborate on why the solution would be involved?

If for all cases we want to return a success instead of an error if we are within planDevMM of the destination this wouldn’t be trivial work. Given where this check takes place currently, we would end up not returning nil,nil which is a pattern go does not allow. Hence we would need to move some things around and rethink of how to return a success to state.go.